plugins/modules/netbox_custom_field.py

```diff
@@ -106,9 +106,30 @@
         required: false
         type: str      
         version_added: "3.10.0"
+      ui_editable:
+         description:
+           - Whether the custom field is editable in the UI
+         required: false
+         choices: 
+           - "yes"
+           - "no"
```

Comment:
These values are going to cause a lot of confusion, people are going to just use `yes` and `no` without strings and YAML will convert them to booleans, and then the module will throw an error and they'll come here asking why it doesn't work.

Reply:
**Exactly** and that is what my comments above are about.  
I 100% agree.

I need feedback on what is the preferred solution to work around this. I do not think forcing users to quote values to *just one* module parameter is reasonable, at all.

Two other possible options:

1. Present 2 different boolean parameters instead of presenting `ui_editable` as a string value: `ui_editable` and `ui_editable_hidden`. Based on the combined value of these 2 parameters decide if give `yes` `no` or `hidden` to NetBox API.
2. Use some other values than `yes` and `no` as values `ui_editable`: module parameters takes other values for example `read-write` and calls NetBox API with `yes`; value `read-only` means `no` and `hidden` stays the same.

Open to other suggestions or feedback on these 2. I had an open discussion item at the netbox project and suggestions to not use `yes` and `no` in the NetBox API were ignored.

Both will make the module API diverge from NetBox API which will be sad.

Reply:
I think Option 1 of the other options is the best one. If we're diverging from the NetBox API we may as well do it the best way we can. Thanks for your work on this @bluikko

Reply:
I also think option 1 is the best solution. I'm not crazy about this tri-boolean that the netbox API is exposing.

Reply:
Will rework this to 2x boolean then.